cpu/stm32l1/periph/i2c.c

```diff
@@ -466,100 +362,105 @@ int i2c_write_regs(i2c_t dev, uint8_t address, uint8_t reg, char *data, int leng
 
 void i2c_poweron(i2c_t dev)
 {
-    switch (dev) {
-#if I2C_0_EN
-        case I2C_0:
-            I2C_0_CLKEN();
-            break;
-#endif
-#if I2C_1_EN
-        case I2C_1:
-            I2C_1_CLKEN();
-            break;
-#endif
-    }
+    RCC->APB1ENR |= (RCC_APB1ENR_I2C1EN << dev);
```

Comment:
@PeterKietzmann Hmm [here](https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32l1/periph/gpio.c#L72) is something similiar to this.

It uses the I2C-Enabling-Mask by left shifting it by the value of dev.

My thought:
`RCC_APB1ENR_I2C1EN`   := 0x00200000   := 0000 0000 0010 0000 .... and
`RCC_APB1ENR_I2C2EN` := 0x00400000 := 0000 0000 0100 0000 .... 
(see: [stm32l1xx.h](http://www.keil.com/dd/docs/arm/st/stm32l1xx/stm32l1xx.h))

for dev  = 0 it's the first one and for dev = 1 it's the second one by shifting RCC_APB1ENR_I2C1EN to left with 1.

If you have another idea how to do this better, please tell me :)